reconcile/terraform_cloudflare_dns.py

```diff
@@ -206,17 +220,18 @@ def create_backend_config(
             bucket_name = tf_state.bucket
             bucket_region = tf_state.region
             break
-
-    if bucket_name and bucket_key and bucket_region:
+    try:
         backend_config = TerraformS3BackendConfig(
             aws_acct_creds["aws_access_key_id"],
             aws_acct_creds["aws_secret_access_key"],
             bucket_name,
             bucket_key,
             bucket_region,
         )
-    else:
-        raise ValueError(f"No state bucket config found for account {aws_acct.name}")
+    except UnboundLocalError:
```

Comment:
Something like:

```python
if all([bucket_name, bucket_key, bucket_region]):
```

Wouldn't work? I am not sure about `UnboundLocalError` usage.

Reply:
No, the problem this is trying to solve is a "UnboundLocalError: local variable 'bucket_name' referenced before assignment" error. 

Is your concern that this is catching a very specific error? I thought about it too but considering `TerraformS3BackendConfig ` is just a data class, and any exception should be raised anyway.

Reply:
I agree that starting to catch `UnboundLocalError` feels a little messy to me. What if you assumed this was about `bucket_name`, but it was that `aws_acct_creds` wasn't set?

Could you just assign the variables before this? Or anything else that wouldn't involve only assigning variables under a certain condition, but then trying to reference them after?

```python
bucket_key = bucket_name = bucket_region = None
```

And then leave the original check, or do what @kwilczynski suggested?

Reply:
Updated. Can't use 
```
if all([bucket_name, bucket_key, bucket_region]):
```
because type threw error about Optional[str]